### ISSUE TITLE
ZC and ZQuest Crash (master)

### ISSUE THREAD

**Issue by ZoriaRPG** [ZQuestClassic/ZQuestClassic#144]
ZQuest crashes on 'reading maps', and ZC crashes as soon as the player enters a filename.

This symptom is from the **base** of Master, before I applied my forward-porting!

I think that it is crashing on loading items.

This _**only affects Release builds**_. Debug builds do not crash!

**Comment by ZoriaRPG** [comment]
Reference: This is where ZQuest halts (crashes) when reading the quest data:

```
OK
Initializing sound driver... OK
initializing graphics
done graphics
Loading Quest: new quest...

Decrypting...okay.
Opening...okay.
Reading Header...okay.
Made in ZQuest 250 Beta 27
Reading Rules...okay.
Reading Strings...okay.
Reading Doors...okay.
```

it does not give any error status. (It just freezes.)

Here are some binaries if you want to try them:
http://timelord.insomnia247.nl/zc/zc_dev/ZC_2.55_Alpha_25SEPT_2017.zip

**Comment by evouga** [comment]
Which quest?

**Comment by ZoriaRPG** [comment]
The in-build 'new'quest crashes.

**Comment by evouga** [comment]
After doors comes dmaps, so the hang must be happening there. I cannot reproduce it in code I compile on my machine... can you run your code in your IDE and pause it when it hangs, and get a stack trace? If not, I'll have to pepper in more al_traces to try to narrow down the place it's getting stuck.

**Comment by ZoriaRPG** [comment]
@evouga Is it possible to do that with a Release build? I thought that only the debug builds could give you stack information from MSVC???

Did you try the binaries that I compiled on your local machine?

I was pretty sure that it was losing its mind as soon as it tried to read items. It could also be some kind of issue with the allegro library, but I fail to see how or what that might be, given that the same lib works in 2.50.x branch compilations.

**Comment by evouga** [comment]
I don't think it's an Allegro issue.

I've pushed some debugging al_traces. What are the last few lines of allegro.log after the program hangs?

**Comment by ZoriaRPG** [comment]
@evouga :

These are the stack traces from ZQuest:
http://timelord.insomnia247.nl/zc/zc_dev/mastercrash1.png
http://timelord.insomnia247.nl/zc/zc_dev/mastercrash2.png

I had to tinker with MSVC to get it to produce the results, and even then, it only wants to do it if I call it as a debugger when ZQuest crashes., ZC also crashes when trying to load a quest. 

http://timelord.insomnia247.nl/zc/zc_dev/mastercrash_zc1.png
http://timelord.insomnia247.nl/zc/zc_dev/mastercrash_zc2.png

Probably fixed now. I was able to patch and compile, before I pulled the new parser stuff, and ZQuest was stable, although all of the CORE module changes to the editors broke double-clicking entries to edit them. (The editors only flicker.) That seems to have been a common issue that you had when migrating to the dialogue_cpy model. :(
